pkg/network/setup/netconf.go

```diff
@@ -74,9 +78,13 @@ func (c *NetConf) Setup(vmi *v1.VirtualMachineInstance, networks []v1.Network, l
 	configState, ok := c.configState[string(vmi.UID)]
 	c.configStateMutex.RUnlock()
 	if !ok {
-		configStateCache := NewConfigStateCache(string(vmi.UID), c.cacheCreator)
+		cache := NewConfigStateCache(string(vmi.UID), c.cacheCreator)
+		configStateCache, err := upgradeConfigStateCache(&cache, networks)
```

Comment:
AFAIU:
New / Migrated VMs has no cache files yet, so the config-state change files with new naming won't contradict any file.

Running VMs enter the network setup on hotplug requests, 
where a new cache file with a new naming (using the network name) will be created.
The new file won't contradict existing files.
(Same goes for migrated VMs from a version prior to the name scheme change)

Why do we need to mind old cache files at all?

I am afraid it will introduce more maintenance that could bite us in the future.
What will happen if we need to change the cache again, would we need to add more to the upgrade logic?

Reply:
I think it is hard to assure the reconcile loop will not enter the setup multiple times before the VMI is in running phase.
The virt-handler could for example be upgraded in the middle of the setup, when the persisted cache has been already created with the old keys.

But we can just use a more basic reason to do this: We are not interested to have a mix of old and new keys. Such a state will require us to always consider old entries for any change we do now and in the future.
By normalizing everything to a single known state, we are free from thinking about it each time.

> I am afraid it will introduce more maintenance that could bite us in the future.

As mentioned above, we actually do it to make sure we do not get bitten each time we change something. The cognitive load of considering that we have both is high.

> What will happen if we need to change the cache again, would we need to add more to the upgrade logic?

This is not different from upgrading a DB scheme. If we need to change something, e.g. different data structure, we will perform the same steps to allow later steps to consume a single option and not consider more than one.